app/src/main/java/org/mozilla/tv/firefox/search/SearchEngineProviderWrapper.kt

```diff
@@ -48,11 +48,15 @@ class SearchEngineProviderWrapper(private val replacements: Map<String, String>)
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     fun updateSearchEngines(searchEngines: SearchEngineList, replacements: Map<String, String>): SearchEngineList {
-        val defaultSearch = searchEngines.default
+        var defaultSearch = searchEngines.default
+
         @Suppress("NAME_SHADOWING") // Defensive copy & mutable
         val searchEngines = mutableListOf<SearchEngine>().apply { addAll(searchEngines.list) }
 
         replacements.forEach { (old, new) ->
+            if (defaultSearch?.identifier == old) {
```

Comment:
If new is not found, what should happen? I assume we should leave the search engine the same instead of replacing it with null because the old engine won't be removed.

Add a test. :)